Xamarin.PropertyEditing.Mac/Controls/Custom/NumericSpinEditor.cs

```diff
@@ -174,106 +205,106 @@ public NumericSpinEditor ()
+				numericEditor.ConstraintTo (this, (n, c) => n.Height == PropertyEditorControl.DefaultControlHeight - 3),
+
+				incrementButton.ConstraintTo (numericEditor, (s, n) => s.Left == n.Right + stepperSpace),
+				incrementButton.ConstraintTo (numericEditor, (s, n) => s.Top == n.Top),
+				incrementButton.ConstraintTo (numericEditor, (s, n) => s.Width == stepperWidth),
+				incrementButton.ConstraintTo (numericEditor, (s, n) => s.Height == stepperTopHeight),
+
+				decrementButton.ConstraintTo (numericEditor, (s, n) => s.Left == n.Right + stepperSpace),
+				decrementButton.ConstraintTo (numericEditor, (s, n) => s.Top == n.Top + stepperTopHeight),
+				decrementButton.ConstraintTo (numericEditor, (s, n) => s.Width == stepperWidth),
+				decrementButton.ConstraintTo (numericEditor, (s, n) => s.Height == stepperBotHeight),
 			});
+
+			PropertyEditorPanel.ThemeManager.ThemeChanged += ThemeManager_ThemeChanged;
+
+			UpdateTheme ();
 		}
 
-		protected virtual void SetStepperActivated ()
+		protected override void Dispose (bool disposing)
 		{
-			SetValue (stepper.DoubleValue);
+			if (disposing) {
+				PropertyEditorPanel.ThemeManager.ThemeChanged -= ThemeManager_ThemeChanged;
+			}
 		}
 
-		protected void OnStepperActivated (object sender, EventArgs e)
+		void ThemeManager_ThemeChanged (object sender, EventArgs e)
 		{
-			if (!editing) {
-				editing = true;
-				SetStepperActivated ();
-				if (ValueChanged != null)
-					ValueChanged (this, EventArgs.Empty);
-				editing = false;
-			}
+			UpdateTheme ();
 		}
 
-		protected void OnValueChanged (object sender, EventArgs e)
+		protected void UpdateTheme ()
 		{
-			if (!editing) {
-				editing = true;
-				SetValue (numericEditor.StringValue);
-				if (ValueChanged != null)
-					ValueChanged (this, EventArgs.Empty);
-				editing = false;
-			}
+			this.Appearance = PropertyEditorPanel.ThemeManager.CurrentAppearance;
 		}
 
-		protected void OnEditingEnded (object sender, EventArgs e)
+		virtual protected void OnEditingEnded (object sender, EventArgs e)
 		{
 			if (!editing) {
 				editing = true;
 				SetValue (numericEditor.StringValue);
-				if (EditingEnded != null)
-					EditingEnded (this, EventArgs.Empty);
-				if (ValueChanged != null)
-					ValueChanged (this, EventArgs.Empty);
+				EditingEnded?.Invoke (this, EventArgs.Empty);
 				editing = false;
 			}
 		}
 
 		void SetValue (string value)
 		{
-			//Regulates maximun and minium out of range
-			if (!string.IsNullOrEmpty (value)) {
-				stepper.DoubleValue = CoerceValue (FieldValidation.FixInitialValue (value, Value.ToEditorString ()).ToEditorDouble ());
-				numericEditor.StringValue = FieldValidation.RoundDoubleValue (stepper.DoubleValue.ToEditorString (), NumericMode == ValidationType.Decimal ? FieldValidation.DefaultXcodeMaxRoundDigits : 0);
-			} else {
-				stepper.StringValue = value;
-				numericEditor.StringValue = value;
-			}
+			numericEditor.StringValue = value;
+			ValueChanged?.Invoke (this, EventArgs.Empty);
```

Comment:
This appears to invoke ValueChanged even if the value hasn't changed?

Reply:
@lewing Do you mean a negative ratios arn't working? If you mean normal float/integer that appears to work for me, as per this screenshot.
<img width="478" alt="screen shot 2018-05-17 at 14 13 49" src="https://user-images.githubusercontent.com/271363/40179586-9b7ec276-59dc-11e8-82c6-a2e53681aa64.png">

Reply:
Does he mean values like `0.5`?

Also, did you mean to include a screenshot or a screencast? :p I do love screencasts :)

Reply:
yeah, I was using the NumericSpinEditor to handle values that must be between 0 and 1, the NSStepper version handled that as expected by limiting to Min/Max and incrementing by the Step increment of .01 when clicking up/down.

Reply:
thanks @lewing I'll double check the original behaviour.

Reply:
@lewing something like this - https://www.screencast.com/t/PenH2m0lU0L ?